net: shell: Add command to show HTTP server information

jukkar opened the pull request:
The new command is "net http" which shows all services and related resources defined in the system.

jukkar committed:
net: shell: Add command to show HTTP server information

The new command is "net http" which shows all services and
related resources defined in the system.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

rlubos commented on subsys/net/lib/shell/http.c:
```diff
@@ -0,0 +1,60 @@
+		   svc->concurrent, svc->backlog);
+
+		HTTP_SERVICE_FOREACH_RESOURCE(svc, res) {
+			PR("\t%s\n", res->resource);
```
Maybe we could fetch the resource detail, and print the type of resource here as well?

jukkar replied:
Good idea. Now the shell prints this
```
Host:Port       Concurrent/Backlog
        Resource type   Methods         Endpoint

192.0.2.1:80    1/10
             dynamic    GET,POST        /dynamic
              static    GET             /
           websocket    GET             /

1 service and 3 resources found.
```

jukkar committed:
net: http: Make sure not use undefined Kconfig symbols

If CONFIG_HTTP_SERVER is not enabled, then we must not use
Kconfig symbols that are only available when HTTP server is
enabled.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>
